tools/src/main/java/org/apache/kafka/trogdor/workload/PartitionsSpec.java

```diff
@@ -44,8 +46,16 @@ public PartitionsSpec(@JsonProperty("numPartitions") int numPartitions,
             @JsonProperty("partitionAssignments") Map<Integer, List<Integer>> partitionAssignments) {
         this.numPartitions = numPartitions;
         this.replicationFactor = replicationFactor;
-        this.partitionAssignments = partitionAssignments == null ?
-            new HashMap<Integer, List<Integer>>() : partitionAssignments;
+        HashMap<Integer, List<Integer>> partMap = new HashMap<>();
+        if (partitionAssignments != null) {
+            for (Entry<Integer, List<Integer>> entry : partitionAssignments.entrySet()) {
+                int partition = entry.getKey() == null ? 0 : entry.getKey();
```

Comment:
Can`entry.getKey()` ever return null? I ask because two lines below we assume that the list `entry.getValue()` doesn't contain any null values and that seems reasonable. I was wondering why this check is required while the other isn't.

Reply:
I think both the map keys and the list can contain null entries.  I will fix it to check in both.